File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        left = 0
        right = 0
        chars = {}
        maxx = 0
        while right < len(s):
            if s[right] in chars:
                left = max(left, chars[s[right]] + 1)
            chars[s[right]] = right
            maxx = max(maxx, right - left + 1)
            right += 1
        return maxx

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0, right = 0;
        int[] chars = new int[256];
        Arrays.fill(chars, -1);
        int maxx = 0;
        while (right < s.length()) {
            if (chars[s.charAt(right)] != -1) {
                left = Math.max(left, chars[s.charAt(right)] + 1);
            }
            chars[s.charAt(right)] = right;
            maxx = Math.max(maxx, right - left + 1);
            right++;
        }
        return maxx;
    }
}
